src/lib/proof-system/zkprogram.ts

```diff
@@ -513,7 +555,7 @@ function ZkProgram<
     get: () => doProving,
   });
 
-  return program as any;
+  return program;
```

Comment:
I worked a bit on types in this file to reduce the amount of `as any`, in particular it's nice to have this final sanity check that the returned program matches the output type. for that, I had to put a boolean into `program.proofsEnabled` which we overwrite later with `Object.defineProperty()`